model.go

```diff
@@ -1225,6 +1284,8 @@ func importModelV1(source map[string]interface{}) (*model, error) {
 func importModelV2(source map[string]interface{}) (*model, error) {
 	fields, defaults := modelV1Fields()
 	fields["remote-applications"] = schema.StringMap(schema.Any())
+	fields["sla"] = schema.StringMap(schema.Any())
```

Comment:
Since we are tying the sla and meter-status formats to the model version, this schema should be fully explicit, not Any.

```go
fields["sla"] = schema.FieldMap(
    schema.Fields{
        "level": schema.String(),
        "credentials":schema.String(),
    }, nil) // no defaults
```

And similarly for the meter-status.

Since you have not versioned the individual SLA and MeterStatus formats, keep the serialization close to the model code, not in a separate file.

Then your constructor funcs can assume a valid map.
```go
func importSLA(source map[string]interface{}) sla {
	return sla{
		Level_:       source["level"].(string),
		Credentials_: source["credentials"].(string),
	}
}
```